src/test/java/com/yahoo/memory/WritableMemoryTest.java

```diff
@@ -54,7 +54,9 @@ public void checkSelfArrayCopy() {
   public void checkEquals() {
     int len = 7;
     WritableMemory wmem1 = WritableMemory.allocate(len);
-    //assertTrue(wmem1.equals(wmem1)); //the plexus compiler w/ errorprone does not like this
+    @SuppressWarnings({"EqualsWithItself", "SelfEquals"})
```

Comment:
These SuppressWarnings tags are unsupported.  And to support them would require adding dependencies to Memory, which I don't want to do.

Reply:
Doing a `javac -X` on JDK8 reveals the following list:

all, auxiliaryclass, cast, classfile, deprecation, dep-ann, divzero, empty, fallthrough, finally, none, options, overloads, overrides, path, processing, rawtypes, serial, static, try, unchecked, varargs.

Beyond that, it is a function of the particular compiler.  I use Eclipse compiler which adds a lot more, but does not include "EqualsWithItself" or "SelfEquals".

I admit I haven't been very consistent in restricting annotations to just the javac set, but it is probably a good idea to do so.

Reply:
I don't get this reasoning. Writing anything like `@SuppressWarnings("randomFooBar")` is harmless. As long as it helps other tools, error-prone (that "officially" participates this build) and IntelliJ, why wouldn't add such annotations?

Reply:
This means I have to disable "Unhandled tokens in @SuppressWarnings" check in Eclipse, which means ignoring misspelled tokens.  I will allow this for now, but I wish we had a better solution.

Reply:
The worst thing that could happen if you overlook an unhandled token in `@SuppressWarnings` is reoccurrence of some warning. When you notice it, you could fix it.